### DISCUSSION TITLE
Rule engine

### DISCUSSION THREAD

**Discussion by ankurcts** [community/community#110897] [Programming Help]
### Body

I have created a rule in the rule engine in the azure blob storage that is Validatealert and this rule defines the expression which contains the left and right side dynamic value but somehow, the right side value creates an issue. So are we supporting right side dynamic value or not? For example-
{
      'RuleName': 'ValidateAlert',
      'Expression':(account.AuthorizationKey.ToString()==(account.AuthorizationKey.ToString()==commonEvent.DynamicAuthorizationKey.ToString()),
'ErrorMessage': 'Failed : Dynamic alert is invalid.'
}

I'm getting the parser issue that is 'Exception while parsing expression Enum type 'DynamicAuthorizationKey' not found'. I tried with multiple permutation and combination to resolve that issue but just want to verify whether microsoft is supporting or not right side dynamic value in the rule engine definition.  

Kindly have a look into this issue, and please suggest to me an appropriate solution.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by P09s** [comment]
Based on the provided code snippet and the error message you're encountering, it seems there's a syntax error in your expression. Azure Blob Storage's rule engine does support dynamic values on both the left and right sides of expressions.

Let's correct the expression and resolve the issue:

```json
{
    'RuleName': 'ValidateAlert',
    'Expression': (account.AuthorizationKey.ToString() == commonEvent.DynamicAuthorizationKey.ToString()),
    'ErrorMessage': 'Failed : Dynamic alert is invalid.'
}
```

In the corrected expression:
- We removed the redundant `(account.AuthorizationKey.ToString()==` part, as it seems unnecessary and possibly causing confusion.
- We adjusted the expression to compare `account.AuthorizationKey` with `commonEvent.DynamicAuthorizationKey`.

Ensure that both `account.AuthorizationKey` and `commonEvent.DynamicAuthorizationKey` are valid and accessible variables or properties in your environment. If they are not being recognized, ensure they are properly defined or imported.

If the issue persists, double-check the documentation for the Azure Blob Storage rule engine or consider reaching out to Microsoft support for further assistance.

Hope this migth help

**Comment by ankurcts** [comment]
Hi Parag,

Yesterday, I had scheduled a meeting with the Microsoft team and the GitHub team, but the GitHub team did not join the meeting.  Please suggest how you should invite the GitHub team to join the meeting.
Kindly give me a particular contact person's name to discuss the issue.

As we know, I'm facing the Rule engine issue and working with the Microsoft team for the last three weeks, but unfortunately, I did not get the solution for that specific issue. Even though I raised the request with GitHub, as per Microsoft team suggestion.

You sent me a code to resolve the issue but already tried that one. It's not working at all.

I look forward to hearing from you . If you have any further concerns, do let me know.

Thanks
Ankur Bajpai
